api/src/client/DynamicsBusinessAddressClient.ts

```diff
@@ -0,0 +1,81 @@
+import { LogWriterType } from "@libs/logWriter";
+import { LicenseSearchAddress } from "@shared/license";
+import axios, { AxiosError } from "axios";
+import { BusinessAddressClient, BusinessIdToAddressesMapping } from "../domain/types";
+
+export const DynamicsBusinessAddressClient = (
+  logWriter: LogWriterType,
+  orgUrl: string
+): BusinessAddressClient => {
+  const getBusinessAddresses = async (
+    accessToken: string,
+    businessId: string
+  ): Promise<LicenseSearchAddress[]> => {
+    const logId = logWriter.GetId();
+    logWriter.LogInfo(`Dynamics Business Address Client - Id:${logId}`);
+
+    return axios
+      .get(
+        `${orgUrl}/api/data/v9.2/rgb_addresses?$select=createdon,rgb_city,rgb_county,rgb_name,rgb_state,rgb_street1,rgb_street2,rgb_typecode,rgb_zip,statecode&$filter=(_rgb_businessid_value eq ${businessId})&$top=50`,
+        {
+          headers: {
+            Authorization: `Bearer ${accessToken}`,
+          },
+        }
+      )
+      .then((response) => {
+        logWriter.LogInfo(
+          `Dynamics Business Address Client - Id:${logId} - Response: ${JSON.stringify(response.data)}`
+        );
+        const activeAddresses = response.data.value.filter(
+          (address: DynamicsApiAddressResponse) => address.statecode === 0
+        );
+        return activeAddresses.map((address: DynamicsApiAddressResponse) => processAddress(address));
+      })
+      .catch((error: AxiosError) => {
+        logWriter.LogError(`Dynamics Business Address Client - Id:${logId} - Error:`, error);
+        throw error.response?.status;
+      });
+  };
+
+  const getMappedBusinessAddressesByIds = async (
+    accessToken: string,
+    businessIds: string[]
+  ): Promise<Awaited<BusinessIdToAddressesMapping>[]> => {
+    const getBusinessAddressesById = async (id: string): Promise<BusinessIdToAddressesMapping> => {
+      return { [id]: await getBusinessAddresses(accessToken, id) };
+    };
+
+    return await Promise.all(
+      businessIds.map(
+        async (id: string): Promise<BusinessIdToAddressesMapping> => await getBusinessAddressesById(id)
+      )
```

Comment:
[sand/dust] If you like, this could be written as 
```
return await Promise.all(businessIds.map(getBusinessAddressesById));
```

which might be easier to read